oidc_provider/lib/utils/token.py

```diff
@@ -71,27 +69,25 @@ def encode_id_token(payload, client):
     Represent the ID Token as a JSON Web Token (JWT).
     Return a hash.
     """
-    keys = get_client_alg_keys(client)
-    _jws = JWS(payload, alg=client.jwt_alg)
-    return _jws.sign_compact(keys)
-
-
-def decode_id_token(token, client):
-    """
-    Represent the ID Token as a JSON Web Token (JWT).
-    Return a hash.
-    """
-    keys = get_client_alg_keys(client)
-    return JWS().verify_compact(token, keys=keys)
+    key = client.client_secret
```

Comment:
This is the same: if the algorithm isn't RS256, then the key is the client's secret, otherwise it's the first private key in the database.
The key is passed to pyjwt so it can sign the jwt.
If the key is an rsa key, it must be passed to pyjwt in a format it understands.
This actually brings up another interesting issue: why have an RSAKey model when it could just be an rsakey that exists as a file on the server. I suspect it was just easier to start this way.